plugins/router/f5/plugin.go

```diff
@@ -0,0 +1,183 @@
+
+		glog.V(4).Infof("Modifying endpoints for %s", poolName)
+
+		// The invocation of PoolHasMember has the side effect that it will
+		// initialise PoolMembers[poolName], which is used below, if necessary.
+		poolExists, err := p.F5Client.PoolExists(poolName)
+		if err != nil {
+			glog.V(4).Infof("F5Client.PoolExists failed: %v", err)
+			return err
+		}
+
+		if !poolExists {
+			glog.V(4).Infof("Skipping update for service \"%s\""+
+				" because no pool \"%s\" exists for the service (no route defined?)",
+				poolName, endpoints.Name)
+			return nil
+		}
+
+		members, err := p.F5Client.GetPoolMembers(poolName)
+		if err != nil {
+			glog.V(4).Infof("F5Client.GetPoolMembers failed: %v", err)
+			return err
+		}
+
+		// We need to keep track of which endpoints already existed in F5 in order to
+		// delete any that no longer exist in the updated set of endpoints.
+		//
+		// It would be really nifty if F5 would just let us PUT the new list to the
+		// pool members resource instead of making a bunch of POST and DELETE
+		// requests, bu-u-ut... it doesn't.
+		needToDelete := map[string]bool{}
+		for member, _ := range members {
+			if members[member] {
+				needToDelete[member] = true
+			}
+		}
+
+		glog.V(4).Infof("Updating F5 with new endpoints...")
+
+		for _, subset := range endpoints.Subsets {
+			for _, addr := range subset.Addresses {
+				for _, port := range subset.Ports {
+					dest := fmt.Sprintf("%s:%d", addr.IP, port.Port)
+					exists := needToDelete[dest]
+					needToDelete[dest] = false
+					if exists {
+						glog.V(4).Infof("  Skipping %s because it already exists.", dest)
+					} else {
+						glog.V(4).Infof("  Adding %s...", dest)
+						err = p.F5Client.AddPoolMember(poolName, dest)
+						if err != nil {
+							glog.V(4).Infof("  Error adding endpoint %s to pool %s: %v",
+								dest, poolName, err)
+						}
+					}
+				}
+			}
+		}
+
+		for member, _ := range needToDelete {
+			if needToDelete[member] {
+				glog.V(4).Infof("  Deleting %s...", member)
+				err = p.F5Client.DeletePoolMember(poolName, member)
+				if err != nil {
+					glog.V(4).Infof("  Error deleting endpoint %s from pool %s: %v",
+						member, poolName, err)
+				}
+			}
+		}
+
+		glog.V(4).Infof("Updated F5 with new endpoints.")
+	}
+
+	return nil
+}
+
+func (p *F5Plugin) HandleRoute(eventType watch.EventType,
+	route *routeapi.Route) error {
+	glog.V(4).Infof("Processing route for service: %v (%v)",
+		route.ServiceName, route)
+
+	// Name of the pool in F5.
+	poolName := fmt.Sprintf("openshift_%s_%s_80",
+		route.Namespace, route.ServiceName)
+
+	// Virtual hostname for policy rule in F5.
+	hostname := route.Host
+
+	switch eventType {
+	case watch.Added, watch.Modified:
+		glog.V(4).Infof("Modifying routes for %s.", poolName)
+
+		err := p.F5Client.CreatePool(poolName)
+		if err != nil {
+			glog.V(4).Infof("Error creating pool %s: %v", poolName, err)
+			return err
+		}
+
+		err = p.F5Client.AddPoolRoute(poolName, hostname)
```

Comment:
Yes.  I would assume that the rules in F5 have some sort of specificity ordering ie. if I define route www.example.com then define www.example.com/test it will first look for a path based route and fall back to www.example.com if no matches are found.